runtime/j9vm/jvm.c

```diff
@@ -3655,35 +3657,59 @@ JVM_LoadLibrary(const char *libName)
 	void *result = NULL;
 	J9JavaVM *javaVM = (J9JavaVM*)BFUjavaVM;
 
-	Trc_SC_LoadLibrary_Entry(libName);
-	if (NULL == javaVM->applicationClassLoader) {
-		J9NativeLibrary *nativeLibrary = NULL;
-		J9InternalVMFunctions *vmFuncs = javaVM->internalVMFunctions;
-		J9VMThread *currentThread = vmFuncs->currentVMThread(javaVM);
-		Assert_SC_notNull(currentThread);
-		vmFuncs->internalEnterVMFromJNI(currentThread);
-		vmFuncs->internalReleaseVMAccess(currentThread);
-		if (vmFuncs->registerBootstrapLibrary(currentThread, libName, &nativeLibrary, FALSE) == J9NATIVELIB_LOAD_OK) {
-			result = (void*)nativeLibrary->handle;
-		}
-		vmFuncs->internalAcquireVMAccess(currentThread);
-		vmFuncs->internalExitVMToJNI(currentThread);
-		Trc_SC_LoadLibrary_BootStrap(libName);
-	} else {
-		PORT_ACCESS_FROM_JAVAVM(javaVM);
-		UDATA handle = 0;
-		UDATA flags = 0;
-		UDATA slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags);
-
-		Trc_SC_LoadLibrary_OpenShared(libName);
-		if (0 != slOpenResult) {
-			slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags | J9PORT_SLOPEN_DECORATE);
-			Trc_SC_LoadLibrary_OpenShared_Decorate(libName);
+#if defined(WIN32)
+	char *libNameConverted = NULL;
+	UDATA libNameLen = strlen(libName);
+	PORT_ACCESS_FROM_JAVAVM(javaVM);
+	UDATA libNameLenConverted = j9str_convert(J9STR_CODE_WINDEFAULTACP, J9STR_CODE_MUTF8, libName, libNameLen, NULL, 0);
+	if (libNameLenConverted > 0) {
+		libNameLenConverted += 1;
+		libNameConverted = j9mem_allocate_memory(libNameLenConverted, OMRMEM_CATEGORY_VM);
+		if (NULL != libNameConverted) {
+			memset(libNameConverted, 0, libNameLenConverted);
```

Comment:
This is unnecessary just to set the last byte.

Reply:
`j9str_convert` null-terminated the string when an extra byte is reserved for `null`.

Please have another look.